substrate/frame/vesting/src/benchmarking.rs

```diff
@@ -383,6 +383,52 @@ benchmarks! {
 		);
 	}
 
+force_remove_vesting_schedule {
+		let l in 0 .. MaxLocksOf::<T>::get() - 1;
+		let s in 2 .. T::MAX_VESTING_SCHEDULES;
+
+		let source: T::AccountId = account("source", 0, SEED);
+		let source_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(source.clone());
+		T::Currency::make_free_balance_be(&source, BalanceOf::<T>::max_value());
+
+		let target: T::AccountId = account("target", 0, SEED);
+		let target_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(target.clone());
+
+		assert!(l <= u8::MAX.into());
+		// Give target existing locks.
+		add_locks::<T>(&target, l as u8);
+
+		let transfer_amount = T::MinVestedTransfer::get();
+		let per_block = transfer_amount.checked_div(&20u32.into()).unwrap();
+
+		let number_of_schedules_after_removal = s - 1;
+		let expected_balance = transfer_amount.checked_mul(&number_of_schedules_after_removal.into()).unwrap();
+
+		// It will remove last vesting schedule.
+		let schedule_index = s - 1;
+
+		let vesting_schedule = VestingInfo::new(
+			transfer_amount,
+			per_block,
+			1u32.into(),
+		);
+
+		for _ in 0..s {
+			assert_ok!(Vesting::<T>::do_vested_transfer(
+				source_lookup.clone(),
+				target_lookup.clone(),
+				vesting_schedule.into()
+			));
+		}
```

Comment:
Seems the benchmark test panics. One quick thing I'd try is to use the `add_vesting_schedules` function in this file as per my [previous comment](https://github.com/paritytech/polkadot-sdk/pull/1982#discussion_r1370037756). It also does top ups of the source account and is just reusable code.

Reply:
yeah I did try this then removed, let's put it back!

Reply:
I just added it and it causes the test to panic, this was why I removed it after trying it out. So I've switched back to
`let expected_balance = transfer_amount.checked_mul(&number_of_schedules_after_removal.into()).unwrap();`